src/main.rs

```diff
@@ -294,6 +200,28 @@ This can be used to provide TLS configuration as in:
     Ok(())
 }
 
+fn to_schema_source(input: Option<&String>, disable_files: bool) -> SchemaSource {
```

Comment:
With the stronger types above, I think this function can change to be "stronger" as well, since parsing URLs can done here

Reply:
What do you mean by stronger here?

Reply:
@mightyshazam What I mean is that the code can actually parse a URL and error out if it cannot be parsed, providing more stringent checks on what the schema source actually is.

I.e. if it's a schema registry that it's a valid URL, if it's a file that the file actually exists (and is a file), not sure if there are other scenarios worth validating but those come to mind